src/components/views/location/LocationPicker.tsx

```diff
@@ -0,0 +1,186 @@
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from 'react';
+import maplibregl from 'maplibre-gl';
+
+import SdkConfig from '../../../SdkConfig';
+import Field from "../elements/Field";
+import DialogButtons from "../elements/DialogButtons";
+import Dropdown from "../elements/Dropdown";
+import { LocationShareType } from "./LocationShareType";
+
+import { _t } from '../../../languageHandler';
+import { replaceableComponent } from "../../../utils/replaceableComponent";
+
+interface IDropdownProps {
+    value: LocationShareType;
+    label: string;
+    width?: number;
+    onChange(type: LocationShareType): void;
+}
+
+const LocationShareTypeDropdown = ({
+    value,
+    label,
+    width,
+    onChange,
+}: IDropdownProps) => {
+    const options = [
+        // <div key={LocationShareType.CUSTOM}>{ _t("Share custom location") }</div>,
+        <div key={LocationShareType.ONE_OFF}>{ _t("Share my current location as a one off") }</div>,
+        // <div key={LocationShareType.ONE_MIN}>{ _t("Share my current location for one minute") }</div>,
+        // <div key={LocationShareType.FIVE_MINS}>{ _t("Share my current location for five minutes") }</div>,
+        // <div key={LocationShareType.THIRTY_MINS}>{ _t("Share my current location for thirty minutes") }</div>,
+        // <div key={LocationShareType.ONE_HOUR}>{ _t("Share my current location for one hour") }</div>,
+        // <div key={LocationShareType.THREE_HOURS}>{ _t("Share my current location for three hours") }</div>,
+        // <div key={LocationShareType.SIX_HOURS}>{ _t("Share my current location for six hours") }</div>,
+        // <div key={LocationShareType.ONE_DAY}>{ _t("Share my current location for one day") }</div>,
+        // <div key={LocationShareType.FOREVER}>{ _t("Share my current location until I disable it") }</div>,
+    ];
+
+    return <Dropdown
+        id="mx_LocationShareTypeDropdown"
+        className="mx_LocationShareTypeDropdown"
+        onOptionChange={(key: string)=>{ onChange(LocationShareType[LocationShareType[parseInt(key)]]); }}
+        menuWidth={width}
+        label={label}
+        value={value.toString()}
+    >
+        { options }
+    </Dropdown>;
+};
+
+interface IProps {
+    onChoose(uri: string, ts: number, type: LocationShareType, description: string): boolean;
+    onFinished();
+}
+
+interface IState {
+    description: string;
+    type: LocationShareType;
+    position: GeolocationPosition;
```

Comment:
```suggestion
    position?: GeolocationPosition;
```

This gets set to undefined in the c'tor